### PR TITLE
Adding FS TC CEPH-11264

### PR DESCRIPTION
**Pull request by AmarnatReddy**
# Description
Adding FS TC CEPH-11264
CEPH-11264 - Analyse the Test Steps planned for Automation
Test Case Flow:
    1. Create 2 buckets with root1 and root2 and move them to root level
    2. Move OSDs of 1 host to root1
    3. Move osds of other host to root2
    4. Collect Crush map
    5. Convert Crush map into text
    6. Edit Crush map and add root1 as destination in replicated rule
    7. Set crush map
    8. Mount and write data to the cephfs
    9. Validate the data going to root1 bucket
    10. Change the crush map and make it to  use root2
    11. Data should be migrated to root2 bucket osds

Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-S4XF0M

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by neha-gangadhar** [comment]
@AmarnatReddy Please add the test ins suite file for executions.

File: tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test Cases Covered:
+    CEPH-11264 - Analyse the Test Steps planned for Automation
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py:17]: Please add appropriate description of the testcase.

File: tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    6. Edit Crush map and add root1 as destination in replicated rule
+    7. Set crush map
+    8. Mount and write data to the cephfs
+    9. Validate the data going to root1 bucket
+    10. Change the crush map and make it to  use root2
+    11. Data should be migrated to root2 bucket osds
+
+    Cleanup:
+    Umount and remove data
+    setb ack the crush to older values
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py:37]: minor change : "set back" correction.

File: tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+            fs_util.create_fs(client1, fs_name)
+            if not fs_util.wait_for_mds_process(client1, fs_name):
+                raise CommandFailed(f"Failed to start MDS deamons for {fs_name}")
+
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd tree -f json")
+        osd_tree = json.loads(out)
+        root_children_ids = [
+            node["children"] for node in osd_tree["nodes"] if node["name"] == "default"
+        ][0]
+        print(root_children_ids)
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py:67]: Modify all print statements using appropriate logger.

File: tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+            f"{df_utilization_root2_step_1['summary']['total_kb_used']}"
+            f"\n After :{df_utilization_root2_step_2['summary']['total_kb_used']}"
+        )
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        pass
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py:282]: Is "pass" necessary here ?

File: tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py [hunk 5]
```diff
@@ -0,0 +1,289 @@
+import json
```
Comment (sumabai) [inline tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py:1]: Is this file required for merge, "tests/cephfs/cephfs_generic/__init__.py"

File: tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd df root1")
+        log.info(f"\n{out}")
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd df root2")
+        log.info(f"\n{out}")
+        if (
+            df_utilization_root1_step_1["summary"]["total_kb_used"]
+            < df_utilization_root1_step_2["summary"]["total_kb_used"]
+        ):
+            raise CommandFailed(
+                f"Data not migrated to the root2 root1 before :"
```
Comment (sumabai) [inline tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py:252]: "Data not migrated from root1 to root2" gives more clarity, also add \n for before and after outputs.

File: tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        fs_util.client_clean_up(
+            "umount", fuse_clients=[clients[0]], mounting_dir=fuse_mounting_dir_1
+        )
+        client1.exec_command(
+            sudo=True, cmd="ceph osd setcrushmap -i crush.map_initial.bin"
+        )
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd tree")
```
Comment (sumabai) [inline tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py:288]: Can the additional OSD pools created be released to ensure no impact on next test? Restoring the original config in OSD and CephFS context?

Comment (AmarnatReddy) [reply tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py:288]: We are not creating any pools. we are using cephfs data pool only.
All OSDs changes will be reverted back when we apply **crush.map_initial.bin** file

**Review by sumabai — APPROVED** [review]

File: tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+        log.info(f"\n{out}")
+        out, rc = client1.exec_command(sudo=True, cmd="ceph osd df root2")
+        log.info(f"\n{out}")
+
+        if (
+            df_utilization_root1_initial["summary"]["total_kb_used"]
+            < df_utilization_root1_step_1["summary"]["total_kb_used"]
+        ):
+            log.info(
+                f"Utilization has been increased after filling data before: "
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_generic/test_data_migrate_bw_buckets.py:193]: Please check, both if and else sections has same logger information.

File: suites/pacific/cephfs/tier-4_cephfs_generic.yaml [hunk 9]
```diff
@@ -0,0 +1,150 @@
+#=======================================================================================================================
+# Tier-level: 4
```
Comment (neha-gangadhar) [inline suites/pacific/cephfs/tier-4_cephfs_generic.yaml:2]: It is the last cephfs_generic component test which is being automated. Does it need a new suite file? Can it be added to any other existing suites considering the conf file used is the generic one?

Comment (AmarnatReddy) [reply suites/pacific/cephfs/tier-4_cephfs_generic.yaml]: Added it tier-4_cephfs_recovery.yaml suite

**Review by neha-gangadhar — APPROVED** [review]
lgtm
